apis/v1alpha1/gateway_types.go

```diff
@@ -234,6 +234,41 @@ type Listener struct {
 	//
 	// Support: Core
 	Routes RouteBindingSelector `json:"routes"`
+
+	// BackendRef is a local object reference to route requests to.
```

Comment:
Can we include some more contextual information here like:
- Setting this backendRef means traffic is directly passed on without any route selection or IOW relationship of this field with `routes` above
- Protocol selection: how is the protocol between the Gateway and this backend selected, this is mentioned in the issue.

Reply:
Could we safely rely on either the protocol specified by the Gateway or the Service? If there are edge cases that wouldn't cover, maybe we'd need an optional `protocol` field in `routeTo`.

Reply:
Thinking about this more, although we may not _need_ a protocol as part of this, I think it would be much easier to understand than any of the alternatives. If we didn't have protocol here, we'd need to read it from the Service spec. That wouldn't work if there were multiple protocols defined for the same port number so we'd likely need to fall back to the protocol defined on the Gateway listener. Adding a protocol field seems like it would be worth it to avoid the potential confusion with the alternative sources.